### DISCUSSION TITLE
how to have different presets with Dive=1 on GoPro13?

### DISCUSSION THREAD

**Discussion by nicklucarellii** [gopro/labs#1162] [Q&A] [answered]
Hi everyone! I wanted to ask if someone could help me create new permanent presets on the GoPro 13 with DIVE=1 mode active. I’d like to create presets with different resolutions and digital lenses so that I can switch presets underwater without using the touchscreen. I saw a reel on YouTube that provided a QR code with the permanent DIVE=1 function, and now I’d like to understand how to create additional permanent DIVE=1 QR codes. Thanks!

**Comment by dnewman-gpsw** [comment]
If you scan *DIVE=1 all Hypersmooth enabled modes / presets will work with a water prescription. This is likely the most reliable. If only same presets are for dive, you can use DIVE in the name. As multiples presets can't be called the same thing, try "DIVE 5K30" and "DIVE 4K60", or similar uppercase four character codes. If the name is a list of four character codes, they we be handed as Labs commands.

**Reply by nicklucarellii** [reply]
Hi, thanks for the reply, but I would like to activate Dive=1 only on specific presets, leaving Hypersmooth normal on others, as in the case of the QR taken from your reels. How can I create other qr like yours, but with different camera settings?

**Reply by dnewman-gpsw** [reply]
As I just explained, all you need to do is have DIVE in the preset name. Within that preset set whatever setting you would like.

**Reply by nicklucarellii** [reply]
ok, so is this command correct? oMPRES="1,DIVE 5KL"*DIVE=1

**Reply by dnewman-gpsw** [reply]
No, that is combining both concepts: preset naming, and global (which is the opposite of what intend.)  All four character codes, must be four characters long.  So try oMPRES="1,DIVE 5KLN"

**Reply by nicklucarellii** [reply]
I tried oMPRES="1,DIVE 5KLN", but the metadata extraction tells me that Dive=1 is not active on the test clip. what are we doing wrong?

**Reply by dnewman-gpsw** [reply]
It works for me, however I see in the code numbers are not support in the 4CC naming so that maybe an issue you are seeing. Try  oMPRES="1,DIVE FVLN"

**Reply by nicklucarellii** [reply]
I tried but this also doesn't show me the Dive=1 mode in the metadata of the test file. I'm creating QR codes on the GoPro QR Code Creator site. I read in other posts that you need to use macros, but I don't know if it has anything to do with this.
<img width="523" alt="Screenshot 2024-11-09 at 10 54 51" src="https://github.com/user-attachments/assets/9000ba55-b9f4-44ba-ac82-14884d388608">

**Reply by dnewman-gpsw** [reply, marked as answer]
You don't need macros, more advanced and you haven't got this working yet. 
Do you get a preset called "DIVE FVLN"?
If you do, select it, you should see the display refresh (black screen for 1/4 second).  DIVE resets the video pipeline to load the new lens prescription. You can see with a careful eye that the stabilization in air is not correct, showing that DIVE mode is active.

The metadata will only be present if you are use this preset.

There is a chance you have to start fresh, maybe something is corrupted in the metadata storage. So if the above doesn't work, scan ```!RESET```, and press shutter three times before the timer runs out.

**Reply by nicklucarellii** [reply]
Yes, I did the reset and now it works! Thank you dear.

**Comment by JiDey0007** [comment]
One more question

Everything works well, but the permanent does not suit me because I must be able to quickly switch to "normal" out of the water

The right mode is therefore my set up plus $WARM=50 for example and the other set up without WARM and everything is good 

My question is that I always use the fast launcher that activates the capture immediately, and the, even with the set up the WARM set is absent, logical the camera to restart

But how does it just block certain set up?

Otherwise I have to leave the camera on the road all the time and the battery doesn't hold

**Reply by dnewman-gpsw** [reply]
Quick Capture mode start up before the Labs service, so you can't combine many Labs features with it.  If you want Labs features active, use the power button, you don't need to leave the camera running.

**Reply by JiDey0007** [reply]
Ok, but it's for spearfishing, the Quick capture is necessary, in the reactive sense but especially easy, without looking I find the button, we don't have time to take our eyes off the fish, turn on, look, start the capture and the fish is already far away

I will use it at the planned times, it will already be good

Ps: for the DIVE it works, I inserted it in the name of the preset and in Quick capture it's ok the DIVE is on 1

**Reply by dnewman-gpsw** [reply]
It does not work with DIVE in the preset name, as quick capture starts before Labs get to read the DIVE name, so you aren't running under water optimized Hypersmooth. I'm surprised you can't have the battery last, how long are your dives?

**Reply by JiDey0007** [reply]
The apneas are chained throughout the day (6-8 hours in the water), and I trigger the video only when an action occurs or to make simple shots (I have several GoPros because I never open the waterproof case on the boat)

Without wanting to contradict you you are an expert and I can't explain but for DIVE, I did the test and passed the file to the data examiner who writes DIVE 1 (After verification I did this test with the 12 that I had just updated)

**Reply by dnewman-gpsw** [reply]
The Metadata is updated, but the enhanced image processing doesn't have time to apply.  The only workaround would be to use a boot command to emulate quick capture. You would use the power button instead, but otherwise.

![image](https://github.com/user-attachments/assets/19ab2d6c-1533-4d21-9e8d-c8c07d205721)

![Quick Capture on Power-11_17_2024, 10_14_54](https://github.com/user-attachments/assets/af155094-6198-4822-aa99-a1f52ecaee93)

While this is 1-2 seconds slower the QC, but all your Labs setting will apply. Now you can use the shutter for above water, and power button for below water, getting the settings you need.

**Reply by JiDey0007** [reply]
Thank you!!

I understand, it's true that I scanned the file but it was late and I didn't look if the image was treated differently

Thank you for this little QR I will try

For the rest if I want to cancel it's !PRESET? Or other ?

**Reply by dnewman-gpsw** [reply]
I don't follow this question: "For the rest if I want to cancel it's !PRESET? Or other ?"

**Reply by JiDey0007** [reply]
I asked for the manipulation if later I want to cancel this new parameter and return to classic

**Reply by dnewman-gpsw** [reply]
Sorry, still not clear to me what you asking.

**Reply by JiDey0007** [reply]
If you don't understand me, it may be me who misunderstood the function of the QR code

Your QR code allows me to launch the record like Quik captures but with this time the labs functions (it works)

My question is, how to restore the original, turn off this QR code if I need (I registered your personal QR)

**Reply by dnewman-gpsw** [reply]
It is a boot command so scan *BOOT="" to remove the boot command from the camera startup. Or just remove the SD, as the boot command is stored on the SD card.

**Reply by JiDey0007** [reply]
Thanks

**Reply by JiDey0007** [reply]
Greeting

I will abandon this idea and I will do the quick spearfishing plan without the DIVE etc.

Too much trouble

The camera was filming it correctly but only turned off with the capture button

The problem is that we can no longer do anything with the camera it captures direct and we really can't even access the video or connect to the Mac

So I scanned boot as you say above and now the lab has disappeared

The launch message is not there and it does not scan

I'm going to sea and it's going to be complicated without the QR codes today 😅

Tonight I would probably do a total RAZ plus install labs again

Unless you tell me something simpler

Thank you anyway

See you later

**Reply by dnewman-gpsw** [reply]
The Labs message was removed with *FAST=1.  This was only to make the power button auto capture near the speed of normal quick capture. This each user has the ability to create your own scripts, you can tailor them to your own needs. I even provided the Blockly example so you can enhance from there. I was only trying to solve the problem you asked about.  If you want use USB (via computer) then add that like this (very simple):

![image](https://github.com/user-attachments/assets/be68e406-726e-45a7-a4e3-07ef25a624f8)
![Quick Capture on Power-11_23_2024, 09_34_25](https://github.com/user-attachments/assets/2d80b2db-835f-400e-9b4c-d7579f42d7d6)

Or just use the SD card for the best speed.

You can make the capture stop with the power button. Basically Labs offers full control, you just need to know what you are asking the camera to do to reap all the benefits.

**Reply by JiDey0007** [reply]
In first I appreciate that you try to help me
In fact no she did not stop, she took the Hilight, only the long support but in the water in fishing action it is not practical

For the rest, there was a quick message I don't know which one, now there is nothing and she doesn't recognize the QRs so I'll update everything

I understood that the interest of the Labs but I don't master well (not yet ..)

I'm going to look at the editor, experiment, it's made for that

The last update 13 with the evolution of red is exactly what I was looking for!
